File: week 1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
  /* initialize pyramid height variable */
  int height = 0;
  /* prompt for input and validate data */
  do
  {
    printf("Height: ");
    height = get_int();
    if (height == 0)
    {
      return 0;
    }
  }
  while (height < 0 || height > 23);
  
  /* loop for number of iterations equal to height of pyramid */
  for (int i = 0; i < height; i++) 
  {
    /* row = i. Loop through number of spaces in row */
    for (int j = 0; j < height - i - 1; j++) 
    {
      printf("%s", " ");
    }
    /* loop through number of hashtags in row starting at row + 1 */
    for (int k = 0; k < i + 1; k++)
    {
      printf("#");
    }
    /* print two spaces between each row */
    printf("%s", "  ");
    /* print hashtags of right pyramid equal to number of hashtags on left side */
    for (int l = 0; l < i + 1; l++)
    {
      printf("#");
    }
    /* print newline to begin new row */
    printf("\n");
  }
}

File: Week 8/mario.py
import cs50

def main():
    
    # Prompt for input height and continue prompting while input height is not between 1 and 23 inclusive
    while True:
        print("Height: ", end="")
        height = cs50.get_int()
        if (height > 0 and height < 24):
            break
    
    # Loop through height value
    for i in range(height):
        # Loop for number of spaces in row (spaces = height - 1 - 1) and print
        for j in range(height-i-1):
            print(" ", end="")
        # Loop for number of hashes in row (hashes = row + 1) and print
        for k in range(i + 1):
            print("#", end="")
        # Print two spaces in the middle of each row
        print("  ", end="")
        # Loop again through number of hases in row to print right side
        for l in range(i+1):
            print("#", end="")
        #Print a new line to move to next row
        print()
        
if __name__ == "__main__":
    main()